pinot-common/src/main/java/org/apache/pinot/common/utils/fetcher/BaseSegmentFetcher.java

```diff
@@ -102,7 +104,8 @@ public void fetchSegmentToLocal(List<URI> uris, File dest)
     });
   }
 
-  public File fetchUntarSegmentToLocalStreamed(URI uri, File dest, long rateLimit)
+  public File fetchUntarSegmentToLocalStreamed(URI uri, File dest, long rateLimit,
+      AtomicInteger attempts)
```

Comment:
We only need the one we are changing, but yes let me look if we can propagate this to count in all uses

Reply:
I looked into bubbling up the attempts in the other calls like fetchSegmentToLocal() but it will not make sense for the metric we are trying to maintain, we would have to define new metrics and at this point we don't need those. These calls are also used from controllers and realtime so they will sprawl metrics to other places so I will skip that.